data/net/minecraft/world/item/enchantment/Enchantment.mapping

```diff
@@ -1,55 +1,103 @@
+		ARG 1 level
 	METHOD getMinLevel ()I
-		COMMENT Returns the minimum level that the enchantment can have.
+		COMMENT Gets the minimum level of this enchantment under normal circumstances.
+	METHOD getOrCreateDescriptionId ()Ljava/lang/String;
+		COMMENT Gets the raw description Id of the enchantment or creates a new one using the registry name of the enchantment.
+		COMMENT @return The descriptionId or a newly created Id.
 	METHOD getRarity ()Lnet/minecraft/world/item/enchantment/Enchantment$Rarity;
-		COMMENT Retrieves the weight value of an Enchantment. This weight value is used within vanilla to determine how rare an enchantment is.
+		COMMENT Gets the rarity of the enchantment. This determines how rare the enchantment will be in enchantment pools such as the enchanting table and loot tables.
 	METHOD getSlotItems (Lnet/minecraft/world/entity/LivingEntity;)Ljava/util/Map;
-		COMMENT Gets map of all the entity's currently equipped gear that this enchantment can go on
-		ARG 1 livingEntity
+		COMMENT Creates a new map containing all items equipped by an entity in slots that the enchantment cares about. {@link slots} These items are not tested for having the enchantment.
+		ARG 1 entity
+			COMMENT The entity to collect equipment for.
 	METHOD isCompatibleWith (Lnet/minecraft/world/item/enchantment/Enchantment;)Z
-		ARG 1 enchantment
+		COMMENT Checks if the enchantment is compatible with another enchantment.
+		COMMENT @return Whether or not both enchantments agree that they are compatible with eachother.
+		ARG 1 other
+			COMMENT The other enchantment to test compatibility with.
+	METHOD isCurse ()Z
+		COMMENT Checks if the enchantment is considered a curse. These enchantments are treated as debuffs and can not be removed from items under normal circumstances.
+		COMMENT @return Whether or not the enchantment is a curse.
 	METHOD isDiscoverable ()Z
-		COMMENT Checks if the enchantment can be applied to loot table drops.
+		COMMENT Checks if the enchantment can be discovered by game mechanics which pull random enchanments from the enchantment registry.
+		COMMENT @return Whether or not the enchantment can be discovered.
 	METHOD isTradeable ()Z
-		COMMENT Checks if the enchantment can be sold by villagers in their trades.
+		COMMENT Checks if the enchantment can be traded by NPCs like villagers.
+		COMMENT @return Whether or not this enchantment can be traded.
+	METHOD isTreasureOnly ()Z
+		COMMENT Checks if the enchantment should be considered a treasure enchantment. These enchantments can not be obtained using the enchantment table. The mending enchantment is an example of a treasure enchantment.
+		COMMENT @return Whether or not the enchantment is a treasure enchantment.
 	CLASS Rarity
+		METHOD <init> (Ljava/lang/String;II)V
+			ARG 3 weight
 		METHOD getWeight ()I
 			COMMENT Retrieves the weight of Rarity.
+		METHOD valueOf (Ljava/lang/String;)Lnet/minecraft/world/item/enchantment/Enchantment$Rarity;
+			ARG 0 name
```

Comment:
This is an effectively synthetic method, no need to document it

Reply:
Should be fine to keep as long as it doesn't break our toolchain?

Reply:
As long as it passes the validator it should be fine.

Reply:
Hmm, this is probably something which we may want to validate andd error for, since to my knowledge this is an artifact of CFR not desugaring enums correctly, as of the current Enigma version that we're using.

Reply:
Ah I see, I will remove this then. We should definitely do something to weed them out.